homeassistant/components/camera/local_file_camera.py

```diff
@@ -0,0 +1,54 @@
+"""Camera that loads a picture from a local file."""
+
+import logging
+import os
+
+from homeassistant.components.camera import Camera
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Camera."""
+    if config.get("file_path") is None:
+        _LOGGER.error("Missing required variable: file_path")
+        return False
+
+    setup_config = (
+        {
+            "name": config.get("name", "Local File Camera"),
+            "file_path": config.get("file_path",
+                                    os.path.join(os.path.dirname(__file__),
+                                                 'image.jpg'))
+        }
+    )
+
+    # check filepath given is readable
+    if not os.access(setup_config["file_path"], os.R_OK):
+        _LOGGER.error("Error: file path is not readable")
+        return False
+
+    add_devices([
+        LocalFileCamera(setup_config)
+    ])
+
+
+class LocalFileCamera(Camera):
+    """Local camera."""
+
+    def __init__(self, device_info):
+        """Initialize Local camera component."""
+        super().__init__()
+
+        self._name = device_info["name"]
+        self._config = device_info
```

Comment:
What about `self._file_path = device_info["file_path"]`? And then use it below in your open statement

Or even passing the name and path into the init instead of the dict?